polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1591,8 +1591,14 @@ mlir::Type CodeGenTypes::getMLIRType(clang::QualType QT, bool *ImplicitRef,
     return Builder.getIntegerType(cast<llvm::IntegerType>(Ty)->getBitWidth());
   }
 
-  if (T->isBuiltinType())
-    return getMLIRType(cast<clang::BuiltinType>(T));
+  if (T->isBuiltinType()) {
+    if (BuiltinTypeCache.find(T) != BuiltinTypeCache.end())
+      return BuiltinTypeCache[T];
```

Comment:
```suggestion
    if (const auto iter = BuiltinTypeCache.find(T); iter != BuiltinTypeCache.end())
      return *iter;
```
Something like this to prevent double lookups.